### DISCUSSION TITLE
Custom `send` method in a TelemetryLogger?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vscode-discussions#522] [Extension Development QnA] [answered]
Hi,
Working on adapting an extension to use the TelemetryLogger API. That class has many methods, like `event` and `error`, and those methods call another method `send(data)` which sends whatever data to a Google Analytics server. Now those methods (`event` and `error`) call `this.telemetryLogger.logUsage`, but there is no way that the TelemetryLogger class knows about the `send` method (other than hardcoding it into both `sendEventData` and `sendErrorData`).
What is the correct way to do this? To be clear, I want to keep the original telemetry class, because it has other useful methods for clean code, but I want that class to now instead use the `TelemetryLogger` API instead of calling `this.send()` directly. Thanks in advance.

**Comment by burkeholland** [comment, marked as answer]
Try this...

Use the TelemetryLogger API in your extension by creating a new method in your class that calls the send() method of the TelemetryLogger class. You can then call this new method from your event() and error() methods instead of calling this.send(). This way, you can keep the original telemetry class and use the TelemetryLogger API at the same time.

Here is an example of how you can create a new method in your class:

```
function sendTelemetry(data) {
    this.telemetryLogger.send({
        eventName: 'myEvent',
        data: data
    });
}
```
You can then call this method from your event() and error() methods like this:

```
function event(data) {
    sendTelemetry(data);
    // Other code here
}

function error(data) {
    sendTelemetry(data);
    // Other code here
}
```
